### ISSUE TITLE
`Series` mapping view

### ISSUE THREAD

**Issue by michaeljpeters** [static-frame/static-frame#987]
**Description**

Calling `.values()` on a Series is raising a TypeError.

However, `.keys()` and `.items()` work as expected.

**Example**

Provide an interactive Python session, or Python code, to demonstrate the issue. Please keep example data to a minimum.

```python
>>> import static_frame as sf
>>>
>>> sf.Series([2,3,4]).values()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'numpy.ndarray' object is not callable
>>>
>>> sf.Series([2,3,4]).keys()
<Index>
0
1
2
<int64>
>>> list(sf.Series([2,3,4]).items())
[(0, 2), (1, 3), (2, 4)]
```

**Platform**

Run the following function (static-frame >= 0.8.1) and provide the results to define your platform and environment:

```python
>>> import static_frame as sf
>>> sf.Platform.display()
<Series: platform>
<Index>
platform           Linux-5.15.0-124-generic-x86_64-with-glibc2.31
sys.version        3.11.4 (main, Aug  1 2024, 09:27:33) [GCC 9.4.0]
static-frame       2.13.0
numpy              1.24.3
pandas             1.5.3
xlsxwriter         3.1.0
openpyxl           3.0.9
xarray             <ModuleNotFoundError>
tables             3.8.0
pyarrow            12.0.1
msgpack            None
msgpack_numpy      None
<<U13>             <object>
```

**Comment by flexatone** [comment]
SF follows the tradition from Pandas of `.values` being a NumPy array attribute, not a function. This is a key difference from the Python Mapping Protocol / `dict` interface, where a `values()` should return `iter(Series.values)`.

If you are a looking for a truly `dict`-compatible interface, we could consider adding a new `SeriesMP`, like the `SeriesHE`, which offers a specialized, completely compatible `Series` interface.

**Comment by michaeljpeters** [comment]
That makes sense for motivation of the `.values` attribute.

For the dict/Mapping interface, maybe instead of creating a full SeriesMP class, there could be a `.via_mapping`/`.as_mapping` that gives back an object that implements the strict Python Mapping Protocol based on a composed Series.

**Comment by michaeljpeters** [comment]
I liked your term "Mapping View". (Even though I think it would be fun to get my initials in sf with `SeriesMP`).

Here's a couple ideas for names:
- `SeriesMappingView`
- `SeriesMV`
- `SeriesMapping`
- `MappingView[Series]`

I noticed that Frame has some similar functionality so maybe it would be worth making this possible in Frame too.

**Comment by flexatone** [comment]
Thanks for these ideas. If the functionality of the view is limited to dict-like operations, I do not think we want to call it a `Series`.

I suppose a similar view for `Frame` is possible as well.

I like the idea of exposing this with a property, like `via_mapping` or `via_mv`; but other options are certainly worth considering.

I will explore an implementation.

**Comment by flexatone** [comment]
Thanks again for these suggestions. `Series.via_mapping` will be available in the 2.14 release.
